Prose/ProseLib/Sources/SidebarFeature/SidebarReducer.swift

```diff
@@ -0,0 +1,73 @@
+import ComposableArchitecture
+import Foundation
+import SharedModels
+
+public typealias UserCredentials = JID
+
+public struct SidebarState: Equatable {
+    var credentials: UserCredentials
+    var selection: Selection?
+    var footer: FooterState
+    var toolbar = ToolbarState()
```

Comment:
I get why you put this here, but one day we'll probably need to start the app composing a new message, and we'll need the "is composing" boollean to be `true`. I guess that's not a problem, and we'll see later. I was just writing as it crossed my mind.

Reply:
Yes, we'll handle that later. There are more problems than that to enable such a feature.